src/entry/bun-worker.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  StartupMessage,
+  ReadyMessage,
+  RequestMessage,
+  ResponseMessage,
+} from '../common'
+import { getHandler, throwInNextTick } from './utils'
+import { stderr, stdout } from 'src/utils'
+
+process.__tinypool_state__ = {
+  isTinypoolWorker: true,
+  isBunWorker: true,
+  workerData: null,
+  workerId: 1,
+}
+
+self.onmessage = onWorkerMessage
+
+function onWorkerMessage(event: MessageEvent<StartupMessage>) {
+  const { filename, name } = event.data
+
+  ;(async function () {
+    if (filename !== null) {
+      await getHandler(filename, name)
+    }
+
+    const readyMessage: ReadyMessage = { ready: true }
+    self.postMessage(readyMessage, '')
+  })().catch(throwInNextTick)
+
+  if (event.ports?.[0]) {
```

Comment:
@sheremet-va I think you mentioned something about Bun and `MessagePort` issues - maybe you run into this one as well:

Bun does not show `event.ports` when printing `event` with `console.log`. Debugging this was really frustrating as I thought `Worker` was unable to receive `MessagePort` from parent. Port doesn't show up on logs but it's still there 🙃 

```js
if (Bun.isMainThread) {
  const workerURL = new URL(import.meta.url).href;
  const worker = new Worker(workerURL);

  const channel = new MessageChannel();
  worker.postMessage("Here is port", [channel.port2]);
} else {
  self.onmessage = (event) => {
    // console.log(event) won't show ports :/
    console.log("[worker] Received event", event);

    // However it is there!
    console.log("\nPort is here:", event.ports);
  };
}
```

<img width="414" alt="image" src="https://github.com/tinylibs/tinypool/assets/14806298/249407a9-9cb1-4dcc-b38c-54ed3c169653">